include/bout/solver.hxx

```diff
@@ -71,7 +71,9 @@ typedef int (*TimestepMonitorFunc)(Solver *solver, BoutReal simtime, BoutReal la
 #include "vector2d.hxx"
 #include "vector3d.hxx"
 
+#define NO_GLOBAL_NAMESPACE
```

Comment:
#1464 will forward declare `PhysicsModel` and remove the include, so presumably this won't be needed after that goes in?

Reply:
With the forward declared `PhysicsModel` the include will move to `solver.cxx` and the `#define`/`#undef` wrapper should go with it.

My intention is to (transparently) let user code keep using `mesh` and `dump` as global names. So:
* put a `using namespace bout::globals` statement in `bout.hxx`
    * `bout.hxx` is included in `physicsmodel.hxx` and `boutmain.hxx` and now nowhere else (`bout.hxx` was also in `difops.cxx` before this PR)
    * User code must include at least one of `bout.hxx`, `physicsmodel.hxx` or `boutmain.hxx`, so get the `using` statement included.
* don't want the library code to have the `using namespace bout::globals` statement so that we ensure that the global mesh pointer is only ever `bout::globals::mesh`
    * this is what the `NO_GLOBAL_NAMESPACE` macro is for - every include of `physicsmodel.hxx` or `bout.hxx` in library code should be wrapped in `#define NO_GLOBAL_NAMESPACE`/`#undef NO_GLOBAL_NAMESPACE` to cut out the `using` statement

It's not great to need the macro, but there are only a couple of places where `physicsmodel.hxx` is included. Still maybe there is a neater way to handle this?